### PR TITLE
LKE-10301: Add the webhooks API

### PR DESCRIPTION
**Pull request by gmasclet**
https://linkurious.atlassian.net/browse/LKE-10301

### CODE REVIEW

File: src/api/Config/types.ts [hunk 1]
```diff
@@ -132,6 +133,12 @@ export interface IGuestPreferencesConfig {
   uiFilter: boolean;
 }
 
+export interface WebhooksConfig {
+  deliveryFrequency?: string;
+  cleanupFrequency?: string;
+  deliveryRetentionDelayMs?: number;
```
Comment (Diachk) [inline src/api/Config/types.ts:139]: Isn't `cleanupFrequency` and `deliveryRetentionDelayMs` the same thing ?

File: src/api/webhook/types.ts [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+  events?: WebhookEvent[];
+}
+
+export interface DeleteWebhookParams {
+  webhookId: number;
+}
+
+export type PingWebhookParams = DeleteWebhookParams;
+
+export interface GetWebhookDeliveriesParams extends DeleteWebhookParams, PaginationClause {}
```
Comment (Diachk) [inline src/api/webhook/types.ts:21]: The fact that multiple interfaces that have nothing to with delete extend the delete params looks weird in my opinion, it makes sense programatically but does not make sense functionally.

I think the cost to make functionally coherent is pretty low.
e.g:
```ts
interface ExistingWebhookParams {
  webhookId: number;
}

export interface DeleteWebhookParams extends ExistingWebhookParams {}

export interface PingWebhookParams extends ExistingWebhookParams {}

export interface GetWebhookDeliveriesParams extends ExistingWebhookParams, PaginationClause {}

```

Bonus point: Adding additional params to `DeleteWebhookParams` will not affect any other interface.

**Review by Diachk — CHANGES_REQUESTED** [review]
Looks good to me overall, i just have a couple of remarks worth analysing.

File: src/api/webhook/types.ts (hunk 2 above)
Comment (gmasclet) [reply src/api/webhook/types.ts]: Good idea, it's done :+1:

File: src/api/Config/types.ts (hunk 1 above)
Comment (gmasclet) [reply src/api/Config/types.ts:139]: * `cleanupFrequency` is a cron expression that controls the schedule of the cleanup task. 
* `deliveryRetentionDelayMs` is the amount of milliseconds after which a delivery becomes eligible for cleanup.

So there not the same thing. For instance, we may schedule the cleanup task once per day at midnight (`0 0 * * *`) and keep completed deliveries for one week (`1000 * 60 * 60 * 24 * 7 => 604800000`).

**Review by Diachk — APPROVED** [review]

**Pull request merged**